neuron.cpp

```diff
@@ -321,6 +321,50 @@ bool Neuron::loadModbusMap()
     return true;
 }
 
+bool Neuron::modbusWriteRequest(QUuid requestId, QModbusDataUnit request)
+{
+    if (!m_modbusInterface)
+        return false;
+
+    if (QModbusReply *reply = m_modbusInterface->sendWriteRequest(request, m_slaveAddress)) {
+        if (!reply->isFinished()) {
+            connect(reply, &QModbusReply::finished, this, [reply, requestId, this] {
+                reply->deleteLater();
+
+                if (!m_writeRequestQueue.isEmpty()) {
```

Comment:
I'd move that down below the requestExecuted() stuff. Because if the next one in the queue is a broadcast request which returns immediately, you'll get replies in a different order than they actually were sent.